subworkflows/local/fasta_windows.nf

```diff
@@ -0,0 +1,77 @@
+//
+// Run fasta_windows and prepare all the output files
+//
+
+include { EXTRACT_COLUMN          } from '../../modules/local/extract_column'
+include { FASTAWINDOWS            } from '../../modules/nf-core/modules/fastawindows/main'
+include { TABIX_BGZIP             } from '../../modules/nf-core/modules/tabix/bgzip/main'
+include { TABIX_TABIX as TABIX_TABIX_CSI   } from '../../modules/nf-core/modules/tabix/tabix/main'
+include { TABIX_TABIX as TABIX_TABIX_TBI   } from '../../modules/nf-core/modules/tabix/tabix/main'
+
+// List of the columns we want to extract as bedGraph from the frequency files,
+// with the subdirectory name and the relevant part of the file name
+ch_freq_config = Channel.of(
+    [4,  'base_content/k1', 'GC'],
+    [5,  'base_content/k1', 'GC_skew'],
+    [6,  'base_content/k1', 'AT_skew'],
+    [7,  'base_content/k1', 'nucShannon'],
+    [8,  'base_content/k1', 'G'],
+    [9,  'base_content/k1', 'C'],
+    [10, 'base_content/k1', 'A'],
+    [11, 'base_content/k1', 'T'],
+    [12, 'base_content/k1', 'N'],
+    [13, 'base_content/k2', 'CpG'],
+    [14, 'base_content/k2', 'dinucShannon'],
+    [15, 'base_content/k3', 'trinucShannon'],
+    [16, 'base_content/k4', 'tetranucShannon'],
+)
+
+// TODO: make it a pipeline-wide parameter ?
+window_size_info = ".1k"
```

Comment:
I agree with this being a pipeline wide parameter, and you can set a default value if needed. I do something similar with `cooler bin size` parameter in `genomenote`.